src/util/types.ts

```diff
@@ -1,45 +1,65 @@
 import * as semver from "semver";
-
 import { Inputs } from "../generated/inputs-outputs";
 
-// https://devblogs.microsoft.com/typescript/announcing-typescript-4-1/#template-literal-types
-// neat
-export type InstallableClient = `${Inputs}`;
+// https://www.typescriptlang.org/docs/handbook/utility-types.html#excludetype-excludedunion
+export type InstallableClient = Exclude<`${Inputs}`, "source" | "github_pat">;
 
 export type ClientsToInstall = { [key in InstallableClient]?: semver.Range };
 
 /**
  * Store here any details for the client that do not match the "expected" values.
+ * GithubRepositoryPath has the repository path of the respective client
  * For example, directoryName usually matches the executable name (the InstallableClient),
  * but if it doesn't, it's overridden here.
+ * isHashMissingOn* stores boolean value that indicates if hash file is missing or not in mirror/github
  */
 export const ClientDetailOverrides: { [key in InstallableClient]?: {
-    directoryName?: string;
+    mirrorDirectoryName?: string;
```

Comment:
i think this makes a little more sense if we group the github / mirror info:

```typescript
mirror?: {
  directoryName: string
  isHashMissing?: true // (there's no reason you'd set this to false since undefined means false. or you can just give it `boolean` type and define it for all)
},
github?: {
  repoSlug: string;
  isHashMissing: ; // same as above
}
```

Reply:
This is good, but I don't think we need `repoSlug` here. This is not used at all

Reply:
Yeah, this is good.